assets/js/components/post-searcher.js

```diff
@@ -112,28 +115,37 @@ class PostSearcher extends Component {
 	}
 
 	onConfirm( selection ) {
-		this.setState( {
-			selection,
-		} );
+		const { results } = this.state;
+		let match;
+		// Check to that the selection is "valid".
+		if ( ! Array.isArray( results ) || ! this.disallowedSelections.includes( selection ) ) {
```

Comment:
This logic is slightly off. `results` must always be an array here because we are going to call `find` on it immediately after this check. We also want to make sure that the selection is not in the selections that are not allowed (i.e. `&&`).

I like the idea of `disallowedSelections`, but since there's only one, and we also have that as a property, I think we can do without it until we have more while still preserving the same readability as it's logical that you shouldn't be able to select the `noResultsMessage`.